wolfcrypt/src/asn.c

```diff
@@ -23502,9 +23507,13 @@ int wc_PemPubKeyToDer_ex(const char* fileName, DerBuffer** der)
             ret = BUFFER_E;
         }
         sz = XFTELL(file);
-        XREWIND(file);
+        if (XFSEEK(file, 0, XSEEK_SET) != 0) {
+            ret = BUFFER_E;
+        }
 
-        if (sz <= 0) {
+        if (ret < 0) {
```

Comment:
Typically we go:
```
if ((ret > 0) && (sz <= 0)) {
```
And don't put 'else' after '}'!

Reply:
did one better and refactored with proper error cascading.